### PR TITLE
access bulk-scan storage via app gateway

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1063

### Change description ###

We can reach the bulk-scan app gateway only by proxy. proxy for bulk_scan introduced.
flux changes:
https://github.com/hmcts/cnp-flux-config/pull/2254

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -94,3 +94,8 @@ scheduling:
       cron: 0 0 6 * * *
     scan:
       delay: ${TASK_SCAN_DELAY} # In milliseconds
+
+
+proxy:
+  host-name: ${PROXY_HOST_NAME:proxyout.reform.hmcts.net}
+  port: ${PROXY_PORT:8080}
```
Comment (doncem) [inline src/main/resources/application.yaml]: perhaps env vars are not needed

File: src/main/java/uk/gov/hmcts/reform/blobrouter/config/HttpConfiguration.java [hunk 2]
```diff
@@ -28,6 +33,23 @@ public HttpComponentsClientHttpRequestFactory clientHttpRequestFactory() {
         return new HttpComponentsClientHttpRequestFactory(getHttpClient());
     }
 
+    @Bean
+    public com.azure.core.http.HttpClient azureHttpClient(
+        @Value("${proxy.host-name}") String proxyHostName,
+        @Value("${proxy.port}") int proxyPort) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/config/HttpConfiguration.java]: style: closing brace should be in new line ;)

File: src/main/resources/application.yaml (hunk 1 above)
Comment (mdayican) [reply src/main/resources/application.yaml]: I added for future use if we need in product chart or for other env..

Comment (doncem) [reply src/main/resources/application.yaml]: if need to override it'll be bootstrapped then. giving the current setup of our apps

File: charts/reform-scan-blob-router/values.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ java:
     TASK_SCAN_DELAY: "4000" # in millis
 
     STORAGE_URL: https://reformscan.{{ .Values.global.environment }}.platform.hmcts.net
-    STORAGE_BULKSCAN_URL: https://bulkscan{{ .Values.global.environment }}.blob.core.windows.net
+    STORAGE_BULKSCAN_URL: https://bulkscan.{{ .Values.global.environment }}.platform.hmcts.net
```
Comment (nitinprabhu) [inline charts/reform-scan-blob-router/values.yaml]: you will need to override this in flux for prod

**Pull request merged**